wfe2/wfe.go

```diff
@@ -1275,3 +1279,74 @@ func (wfe *WebFrontEndImpl) addIssuingCertificateURLs(response http.ResponseWrit
 	}
 	return nil
 }
+
+type orderJSON struct {
+	Status         core.AcmeStatus
+	Expires        time.Time
+	CSR            core.JSONBuffer
+	Authorizations []string
+}
+
+// NewCertificate is used by clients to create a new order object from a CSR
+func (wfe *WebFrontEndImpl) NewOrder(ctx context.Context, logEvent *requestEvent, response http.ResponseWriter, request *http.Request) {
+	body, reg, prob := wfe.validPOSTForAccount(request, ctx, logEvent)
+	addRequesterHeader(response, logEvent.Requester)
+	if prob != nil {
+		// validPOSTForAccount handles its own setting of logEvent.Errors
+		wfe.sendError(response, logEvent, prob, nil)
+		return
+	}
+
+	var rawCSR core.RawCertificateRequest
+	err := json.Unmarshal(body, &rawCSR)
```

Comment:
Could you add a note here that the `NotAfter` and `NotBefore` fields of the order request are discarded? They're marked OPTIONAL in [Section 7.4](https://tools.ietf.org/html/draft-ietf-acme-acme-07#section-7.4) but I think it would help readability/client devs if you call out explicitly that we discard those fields of new-order requests here.